src/imagery/i.asf.download/i.asf.download.py

```diff
@@ -0,0 +1,213 @@
+# MODULE:       i.sentinel_1.download_asf
+#
+# AUTHOR(S):    Guido Riembauer
+#
+# PURPOSE:      Searches and Downloads SAR data from the Alaska Satellite Facility
+#
+#
+# COPYRIGHT:	(C) 2020-2022 by mundialis GmbH & Co. KG and the GRASS Development Team
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#############################################################################
+
+# %Module
+# % description: Searches and downloads SAR data from the Alaska Satellite Facility.
+# % keyword: imagery
+# % keyword: satellite
+# % keyword: download
+# % keyword: SAR
+# % keyword: Sentinel
+# %end
+
+# %option
+# % key: output_directory
+# % type: string
+# % required: no
+# % description: Name for output directory where to store downloaded Sentinel data
+# % label: Directory where to store downloaded data
+# %end
+
+# %option
+# % key: aoi
+# % type: string
+# % required: yes
+# % multiple: no
+# % description: Path to aoi GeoJSON file
+# % label: Path to aoi GeoJSON file
+# %end
+
+# %option
+# % key: token
+# % type: string
+# % required: yes
+# % multiple: no
+# % description: Path to ASF token file
+# % label: File has to contain ASF token
+# %end
+
+# %option
+# % key: start
+# % required: no
+# % type: string
+# % description: Start date ('YYYY-MM-DD')
+# % guisection: Filter
+# %end
+
+# %option
+# % key: end
+# % required: no
+# % type: string
+# % description: End date ('YYYY-MM-DD')
+# % guisection: Filter
+# %end
+
+# %option
+# % key: platform
+# % required: yes
+# % type: string
+# % description: Satellite platform
+# % label: Currently only Sentinel-1 is supported
+# % options: Sentinel-1
+# % answer: Sentinel-1
+# %end
+
+# %option
+# % key: beamMode
+# % required: yes
+# % type: string
+# % description: Satellite beam mode
+# % label: Currently only IW is supported
+# % options: IW
+# % answer: IW
+# %end
+
+# %option
+# % key: processinglevel
+# % required: no
+# % type: string
+# % description: SAR processing level
+# % label: Ground-Range-Detected (GRD) or Single-Look-Complex (SLC)
+# % options: GRD_HD,SLC
+# % answer: GRD_HD
+# %end
+
+# %flag
+# % key: l
+# % description: Only list scenes available
+# %end
+
+# %flag
+# % key: c
+# % description: Do checksum test of all scenes
+# %end
+
+# %rules
+# % required: output_directory,-l
+# % required: start
+# % collective: start,end
+# %end
+
+import zipfile
+import os
+import json
+import hashlib
+import concurrent.futures
+
+from pathlib import Path
+
+import shapely.wkt
+import asf_search as asf
+import geopandas as gpd
```

Comment:
```suggestion
```
These thirds party libraries should be imported as "lazy imports" (at the bottom)...